### PR TITLE
Updating template text readme.md

### PR DESCRIPTION
**Pull request by jackiekazil**
This is a simple update to the readme that describes templates and how to create them.

### CODE REVIEW

**Comment by jackiekazil** [comment]
@jeetu7 can you give this a quick review? 
Also, based on your PR request #272, can you also suggest a change for [line 39](https://github.com/projectmesa/mesa/pull/273/files#diff-97b6d846aeebd09f87a3d9665ca77fd8R39)

File: mesa/visualization/ModularVisualization.md [hunk 1]
```diff
@@ -1,32 +1,31 @@
...
+Under the hood, each visualization module consists of two parts:
 
-Okay, so what do you actually need to know?
+1. **Data rending** - Python code which can take a model object and renders it into some data describing the visualization.
+2. **Browser rendering** - JavaScript object which in turn receives data from the Python render (via the ModularServer) and actually draws it in the browser.
 
 ## Using Pre-Built Modules
 
-Most of the time, you should be able to build your visualization using pre-built modules. To do this, you don't need to worry about HTML and JavaScript at all, and only look at the documentation for the particular module you want to use.
+Part of the Mesa library are pre-built modules. Using the built-ins allow you to build a visualization without worrying about the HTML and javascript. Consult the documention for a variety of modules.
```
Comment (jeetu7) [inline mesa/visualization/ModularVisualization.md]: First sentence on Line 26 can be replaced with this.
"Mesa already comes with some pre-built modules."

File: mesa/visualization/ModularVisualization.md [hunk 2]
```diff
@@ -67,25 +66,25 @@ Then, to launch a server with this grid as the only visualization element:
...
             Args:
                 attr_name: The name of the attribute to extract from the model.
             '''
             self.attr_name = attr_name
-    
-        def render(self, model): 
+
+        def render(self, model):
             val = getattr(model, self.attr_name)
             return attr_name + ": " + str(val)
```
Comment (jeetu7) [inline mesa/visualization/ModularVisualization.md]: On line 89, what will attr_name expand to?

File: mesa/visualization/ModularVisualization.md [hunk 3]
```diff
@@ -67,25 +66,25 @@ Then, to launch a server with this grid as the only visualization element:
 ```
 ## Sub-Classing Modules
 
-In some cases, you may want or need to tweak the internals of an existing visualization module. The best way to do this is to create a subclass of it.
+If you want to tweak the internals of an existing visualization module, try subclassing it.
```
Comment (jeetu7) [inline mesa/visualization/ModularVisualization.md]: Line 69: You might prefer something like a mix of your's and orignal's:

"In some cases, you may want to customize the internals of an existing visualization module. The best way to do this is to create a subclass of it."

**Comment by jeetu7** [comment]
For line 39 Check my commit's ModularVisualization.md . 
Hope this helps.

**Comment by jeetu7** [comment]
If possible, we can also stick to 79 characters limit in .md files as well (as far as possible till it does not mess up with the code example part).

**Comment by jackiekazil** [comment]
@jeetu7 I went through all your suggestions, let me know if anything doesn't look clear. 
RE: 79 chars -- I am not sure I understand. Is this for code & for text? If just code, which lines violate this? Everything appears to be under 80 chars in my code editor.

**Comment by jackiekazil** [comment]
Also, I didn't touch line 39, because I didn't want to end up with a merge conflict.

**Comment by jeetu7** [comment]
@jackiekazil 80 char limit for text itself.

**Comment by jackiekazil** [comment]
@jeetu7 at this point, I don't see the value add to do that. While it might not effect the formatting, it will make writing a lot more difficult and lead to abrupt hanging lines. I don't think I have ever see other projects have this kind of enforcement. (If you show me a best practice for it, I will accept PRs that do this.)

**Comment by jackiekazil** [comment]
I am going to merge this in to close out.

**Pull request merged**